contracts/factory/NexusAccountFactory.sol

```diff
@@ -42,14 +42,17 @@ contract NexusAccountFactory is Stakeable, INexusFactory {
     /// @param salt Unique salt for the Smart Account creation.
     /// @return The address of the newly created Nexus account.
     function createAccount(bytes calldata initData, bytes32 salt) external payable override returns (address payable) {
-        // Compute the actual salt for deterministic deployment
+        // Compute the actual salt as the hash of initData and salt using assembly
         bytes32 actualSalt;
         assembly {
+            // Load the free memory pointer
             let ptr := mload(0x40)
-            let calldataLength := sub(calldatasize(), 0x04)
-            mstore(0x40, add(ptr, calldataLength))
-            calldatacopy(ptr, 0x04, calldataLength)
-            actualSalt := keccak256(ptr, calldataLength)
+            // Store initData at the free memory pointer
+            calldatacopy(ptr, initData.offset, initData.length)
```

Comment:
why not consistent everywhere?
here it would be
bytes32 actualSalt = keccak256(abi.encodePacked(initData, salt))

Reply:
that's what it does but in assembly

Reply:
I mean, why cant we just use this abi.encodePacked way everywhere then.

Reply:
like in other factories we have changed and used like this
bytes32 actualSalt = keccak256(abi.encodePacked(eoaOwner, index, attesters, threshold));

go back to the issue please to also understand test case needs.
we are still hashing everything for salt event without any change, but in the event it would be emitted different value which won't help to reconstruct salt

Reply:
@Aboudjem 
cc @MrToph